File: src/main/java/de/storchp/opentracks/osmplugin/maps/MovementDirection.java
package de.storchp.opentracks.osmplugin.maps;

import org.oscim.core.GeoPoint;

import de.storchp.opentracks.osmplugin.utils.MapUtils;

public class MovementDirection {

    private GeoPoint secondToLastPos;
    private float currentDegrees = 0;

    public void updatePos(GeoPoint endPos) {
        if (endPos != null && !endPos.equals(secondToLastPos)) {
            currentDegrees = MapUtils.bearingInDegrees(secondToLastPos, endPos);
            secondToLastPos = endPos;
        }
    }

    public float getCurrentDegrees() {
        return currentDegrees;
    }

}

File: src/main/kotlin/de/storchp/opentracks/osmplugin/maps/MovementDirection.kt
package de.storchp.opentracks.osmplugin.maps

import de.storchp.opentracks.osmplugin.utils.MapUtils
import org.oscim.core.GeoPoint

class MovementDirection {
    private var secondToLastPos: GeoPoint? = null
    var currentDegrees = 0f
        get() = currentDegrees

    fun updatePos(endPos: GeoPoint?) {
        if (endPos != null && endPos != secondToLastPos) {
            currentDegrees = MapUtils.bearingInDegrees(secondToLastPos, endPos)
            secondToLastPos = endPos
        }
    }

}
